src/TRoomDB.cpp

```diff
@@ -670,7 +673,59 @@ void TRoomDB::auditRooms(QHash<int, int>& roomRemapping, QHash<int, int>& areaRe
         }
     }
 
-    // START OF TASK 5
+    // START OF TASK 5.0 (previously omitted) - add in areas that we have an id
+    // for but no actaul TArea:
+    // * id WILL be in validUsedAreaIds
+    // * id WILL NOT be in areaRemapping, but represents a id number that cannot be
+    // remapped as a new area id
+    // * id MAY be in TRoomDB::areaNamesMap
+    // * id MAY be used by rooms which will have to be added to it at some point
+    // * WILL NOT be in TRoomDB::areas
+    if (!missingAreasNeeded.isEmpty()) {
+        if (mudlet::self()->getAuditErrorsToConsoleEnabled()) {
+            QString alertMsg = tr( "[ ALERT ] - %n area(s) detected as missing in map: adding it/them in.\n"
+                                   " Look for further messsages related to the rooms that are supposed to\n"
+                                   " be in this/these area(s)...",
+                                   "Making use of %n to allow quantity dependent message form 8-) !",
+                                   missingAreasNeeded.count() );
+            mpMap->postMessage(alertMsg);
+        }
+        mpMap->appendErrorMsgWithNoLf(tr("[ ALERT ] - %n area(s) detected as missing in map: adding it/them in.\n"
+                                         " Look for further messsages related to the rooms that is/are supposed to\n"
+                                         " be in this/these area(s)...",
+                                         "Making use of %n to allow quantity dependent message form 8-) !",
+                                         missingAreasNeeded.count() ),
+                              true);
+
+        QString infoMsg;
+        if (mudlet::self()->getAuditErrorsToConsoleEnabled()) {
+            infoMsg = tr("[ INFO ]  - The missing area(s) are now called:\n"
+                         "(ID) ==> \"name\"",
+                         "Making use of %n to allow quantity dependent message form 8-) !",
+                         missingAreasNeeded.count() );
+        }
+
+        QMutableListIterator<int> itMissingArea(missingAreasNeeded);
+        while(itMissingArea.hasNext()) {
```

Comment:
I guess so, I do like the Qt iterators though this type might be fractionally slower.